### PR TITLE
[ICD] Change IdleModeInterval Type from milliseconds to seconds

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### description
SDK update to change IdleModeInterval type from milliseconds to seconds.

Implements Spec and Test Plan changes
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7438
https://github.com/CHIP-Specifications/chip-test-plans/pull/3354

### CODE REVIEW

File: src/app/icd/IcdManagementServer.h [hunk 1]
```diff
@@ -66,9 +66,9 @@ class IcdManagementServer
     IcdManagementServer() = default;
 
     static IcdManagementServer mInstance;
-    uint32_t mIdleInterval           = CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL;
-    uint32_t mActiveInterval         = CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL;
-    uint16_t mActiveThreshold        = CHIP_CONFIG_ICD_ACTIVE_MODE_THRESHOLD;
+    uint32_t mIdleInterval           = CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL;    // in seconds
```
Comment (jmartinez-silabs) [inline src/app/icd/IcdManagementServer.h:69]: can you add a static assert that `CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL <= 64800`

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/icd/ICDManager.cpp [hunk 2]
```diff
@@ -123,8 +123,9 @@ void ICDManager::UpdateOperationState(OperationalState state)
 
     if (state == OperationalState::IdleMode)
     {
-        mOperationalState         = OperationalState::IdleMode;
-        uint32_t idleModeInterval = IcdManagementServer::GetInstance().GetIdleModeInterval();
+        mOperationalState = OperationalState::IdleMode;
+        uint32_t idleModeInterval =
+            (IcdManagementServer::GetInstance().GetIdleModeInterval() * 1000); // convert IdleModeInterval from s to ms
         DeviceLayer::SystemLayer().StartTimer(System::Clock::Timeout(idleModeInterval), OnIdleModeDone, this);
```
Comment (jtung-apple) [inline src/app/icd/ICDManager.cpp:129]: I would suggest just using `System::Clock::Seconds32(idleModeInterval)` in the StartTimer call so the code documents itself, and avoids time arithmetic.

**Review by jtung-apple — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/icd/ICDManager.cpp (hunk 2 above)
Comment (mkardous-silabs) [reply src/app/icd/ICDManager.cpp]: done

File: src/app/icd/IcdManagementServer.h (hunk 1 above)
Comment (mkardous-silabs) [reply src/app/icd/IcdManagementServer.h]: done

**Pull request merged**